packages/common-lib/src/services/worksheetRegistryService.js

```diff
@@ -0,0 +1,182 @@
+  questionSets: 'questionSets',
+  outcomeDeclaration: 'outcomeDeclaration',
+  outcomeProcessing: 'outcomeProcessing',
+  questionSetType: 'questionSetType',
+  criteria: 'criteria',
+  usedFor: 'usedFor',
+  description: 'purpose',
+  visibility: 'visibility',
+  qumlVersion: 'qumlVersion'
+}
+
+let only = Object.keys(interfaceData)
+
+export const getAll = async ({ limit, ...params } = {}, header = {}) => {
+  let headers = {
+    ...header,
+    headers: {
+      ...header.header,
+      Authorization: 'Bearer ' + localStorage.getItem('token')
+    }
+  }
+  const result = await post(
+    manifest.api_url + '/worksheet/search',
+    { filters: params, limit: limit },
+    {
+      headers: headers?.headers ? headers?.headers : {}
+    }
+  )
+
+  if (result.data.data) {
+    return result.data.data.map((e) => mapInterfaceData(e, interfaceData))
+  } else {
+    return []
+  }
+}
+
+export const getOne = async (filters = {}, header = {}) => {
+  let headers = {
+    ...header,
+    Authorization: 'Bearer ' + localStorage.getItem('token')
+  }
+  try {
+    const result = await get(manifest.api_url + '/worksheet/' + filters.id, {
+      headers
+    })
+    if (result?.data?.data) {
+      let mapResult = mapInterfaceData(result.data.data, interfaceData)
+      mapResult.id = mapResult.id?.startsWith('1-')
+        ? mapResult.id?.replace('1-', '')
+        : mapResult.id
+      return mapResult
+    } else {
+      return {}
+    }
+  } catch {
+    return {}
+  }
+}
+
+export const create = async (data, header = {}) => {
+  let headers = {
+    ...header,
+    headers: {
+      ...header.header,
+      Authorization: 'Bearer ' + localStorage.getItem('token')
+    }
+  }
+  let newInterfaceData = interfaceData
+  newInterfaceData = {
+    ...interfaceData,
+    removeParameter: headers?.removeParameter ? headers?.removeParameter : [],
+    onlyParameter: headers?.onlyParameter ? headers?.onlyParameter : only
+  }
+  let newData = mapInterfaceData(data, newInterfaceData, true)
+  const result = await post(manifest.api_url + '/worksheet', newData, {
+    headers: headers?.headers ? headers?.headers : {}
+  })
+  if (result.data) {
+    return true
+    // return result.data.map((e) => mapInterfaceData(e, interfaceData));
+  } else {
+    return false
+  }
+}
+
+export const update = async (data = {}, headers = {}) => {
+  let newInterfaceData = interfaceData
+  newInterfaceData = {
+    ...interfaceData,
+    removeParameter: headers?.removeParameter ? headers?.removeParameter : [],
+    onlyParameter: headers?.onlyParameter ? headers?.onlyParameter : only
+  }
+  let newData = mapInterfaceData(data, newInterfaceData, true)
+
+  const result = await coreUpdate(
+    manifest.api_url + '/worksheet/' + data.id,
+    newData,
+    {
+      headers: headers?.headers ? headers?.headers : {}
+    }
+  )
+  if (result.data) {
+    return result
+  } else {
+    return {}
+  }
+}
+
+export const getAllQuestions = async (filter, request) => {
```

Comment:
Should it be moved out from worksheet service and create separate service for questionsService?